salt/modules/saltcheck.py

```diff
@@ -517,7 +527,11 @@ def run_state_tests(state, saltenv=None, check_all=False, only_fails=False):
         # If passed a duplicate state, don't overwrite with empty res
         if not results.get(state_name):
             results[state_name] = results_dict
-    return _generate_out_list(results, only_fails=only_fails)
+
+        if junit and HAS_JUNIT:
+            return _generate_junit_out_list(results)
+        else:
+            return _generate_out_list(results, only_fails=False)
```

Comment:
**issue (blocking):** I think this was set to `False` during testing and made its may into the PR...

```suggestion
            return _generate_out_list(results, only_fails=only_fails)
```